### PR TITLE
Add the sparseml callback

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Introduces a [SparseML](https://docs.neuralmagic.com/sparseml/) Callback to bolts!

## Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? \[not needed for typos/docs\]
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/callbacks/sparseml.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+To choose a recipe, have a look at `recipes <https://docs.neuralmagic.com/sparseml/source/recipes.html>`__ and `Sparse Zoo <https://docs.neuralmagic.com/sparsezoo/>`__.
+
+It may be easier to infer a recipe via the UI dashboard using `Sparsify <https://github.com/neuralmagic/sparsify>`__ which allows you to tweak and configure a recipe.
+This requires to import an ONNX model, which you can get from your ``LightningModule`` by doing ``model.to_onnx(output_path)``.
+
+2. Train with SparseMLCallback
+------------------------------
+
+.. code-block:: python
```
Comment (Borda) [inline docs/source/callbacks/sparseml.rst:26]:
```suggestion
.. testcode::
```

File: docs/source/callbacks/sparseml.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        callbacks=SparseMLCallback(recipe_path='recipe.yaml')
+    )
+
+3. Export to ONNX!
+------------------
+
+Using the helper function, we handle any quantization/pruning internally and export the model into ONNX format.
+Note this assumes either you have implemented the property ``example_input_array`` in the model or you must provide a sample batch as below.
+
+.. code-block:: python
```
Comment (Borda) [inline docs/source/callbacks/sparseml.rst:44]:
```suggestion
.. testcode::
```

File: docs/source/callbacks/sparseml.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    import torch
+
+    # export the onnx model, using the `model.example_input_array`
+    SparseMLCallback.export_to_sparse_onnx(model, 'onnx_export/')
```
Comment (Borda) [inline docs/source/callbacks/sparseml.rst:49]: plus add a sample model to be able to execute it

**Review by tchaton — APPROVED** [review]
Looks dope !

File: docs/source/callbacks/sparseml.rst [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+2. Train with SparseMLCallback
+------------------------------
+
+.. testcode::
+
+    from pytorch_lightning import LightningModule, Trainer
+    from pl_bolts.callbacks import SparseMLCallback
+
+
+    model = MyModel()
```
Comment (Borda) [inline docs/source/callbacks/sparseml.rst:32]:
```suggestion
    model = nn.Modele()
```
or define a kind of void Model or Boring model?

File: docs/source/callbacks/sparseml.rst [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+------------------
+
+Using the helper function, we handle any quantization/pruning internally and export the model into ONNX format.
+Note this assumes either you have implemented the property ``example_input_array`` in the model or you must provide a sample batch as below.
+
+.. testcode::
+
+    import torch
+
+    model = MyModel()
```
Comment (Borda) [inline docs/source/callbacks/sparseml.rst:48]: is the `MyModel`  defined somewhere in docs?

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

**Review by awaelchli — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Pull request merged**